functions.php

```diff
@@ -144,14 +144,16 @@ function wp_autoupdates_prepare_themes_for_js( $prepared_themes ) {
 		return $prepared_themes;
 	}
 
-	$wp_auto_update_themes = get_option( 'wp_auto_update_themes', array() );
+	$auto_updates = get_option( 'wp_auto_update_themes', array() );
 	foreach ( $prepared_themes as &$theme ) {
 		// Set extra data for use in the template.
 		$slug         = $theme['id'];
-		$encoded_slug = urlencode( $slug );
+		$auto_update  = in_array( $slug, $auto_updates, true );
 
-		$theme['autoupdate']            = in_array( $slug, $wp_auto_update_themes, true );
-		$theme['actions']['autoupdate'] = current_user_can( 'update_themes' ) ? wp_nonce_url( admin_url( 'themes.php?action=autoupdate&amp;theme=' . $encoded_slug ), 'updates' ) : null;
+		$theme['autoupdate']            = $auto_update;
+		$theme['actions']['autoupdate'] = current_user_can( 'update_themes' )
+			? wp_nonce_url( admin_url( sprintf( 'themes.php?action=%s&theme=%s', $auto_update ? 'disable-auto-update' : 'enable-auto-update', urlencode( $slug ) ) ), 'updates' )
+			: null;
```

Comment:
For readability's sake nested ternaries should be avoided as much as possible (this is mentioned in the WP coding style as overly "clever code"). The above three lines would probably be more readable/easier to comprehend if that was an `if - elseif - else` block.

Reply:
Also agree here.  I struggled with that line.

Defining the action string in a variable is closer to the equiv code in core, will push another commit with that change.